jOOQ-codegen-maven/pom.xml

```diff
@@ -103,26 +93,20 @@
             <groupId>org.jooq</groupId>
             <artifactId>jooq-codegen</artifactId>
         </dependency>
-
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-core</artifactId>
-
-            <!-- junit 4.8.2 sneaking in here -->
-            <exclusions>
-                <exclusion>
-                    <groupId>junit</groupId>
-                    <artifactId>junit</artifactId>
-                </exclusion>
-            </exclusions>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.maven.plugin-tools</groupId>
             <artifactId>maven-plugin-annotations</artifactId>
+            <scope>provided</scope>
```

Comment:
These various `provided` scopes don't really look correct to me? What is the rationale? If someone wants to build a plugin based on the jOOQ plugin, wouldn't the jOOQ plugin rather already expose its transitive dependencies, formally?

Reply:
These provided deps are provided by Maven when plugin runs. If you don't put them in provided scope, maven will download them but not use them (as provided ones replaces them). Also, maven 3.9.x due this warns when plugin is executed

Reply:
Can you show a reproducer / log output?

Reply:
OK, I can see it now *after* the dependency upgrades:

```
15:16:29,163 [WARNING]

Some dependencies of Maven Plugins are expected to be in provided scope.
Please make sure that dependencies listed below declared in POM
have set '<scope>provided</scope>' as well.

The following dependencies are in wrong scope:
 * org.apache.maven:maven-plugin-api:jar:3.9.2:compile
 * org.apache.maven:maven-model:jar:3.9.2:compile
 * org.apache.maven:maven-artifact:jar:3.9.2:compile
 * org.apache.maven:maven-core:jar:3.9.2:compile
 * org.apache.maven:maven-settings:jar:3.9.2:compile
 * org.apache.maven:maven-settings-builder:jar:3.9.2:compile
 * org.apache.maven:maven-builder-support:jar:3.9.2:compile
 * org.apache.maven:maven-repository-metadata:jar:3.9.2:compile
 * org.apache.maven:maven-model-builder:jar:3.9.2:compile
 * org.apache.maven:maven-resolver-provider:jar:3.9.2:compile
```